demos/ota/ota_demo_core_mqtt/ota_demo_core_mqtt.c

```diff
@@ -0,0 +1,1354 @@
+
+    if( sem_trywait( &bufferSemaphore ) == 0 )
+    {
+        for( ulIndex = 0; ulIndex < otaconfigMAX_NUM_OTA_DATA_BUFFERS; ulIndex++ )
+        {
+            if( eventBuffer[ ulIndex ].bufferUsed == false )
+            {
+                eventBuffer[ ulIndex ].bufferUsed = true;
+                pFreeBuffer = &eventBuffer[ ulIndex ];
+                break;
+            }
+        }
+
+        ( void ) sem_post( &bufferSemaphore );
+    }
+    else
+    {
+        LogError( ( "Failed to get buffer semaphore: "
+                    ",errno=%s",
+                    strerror( errno ) ) );
+    }
+
+    return pFreeBuffer;
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief The OTA agent has completed the update job or it is in
+ * self test mode. If it was accepted, we want to activate the new image.
+ * This typically means we should reset the device to run the new firmware.
+ * If now is not a good time to reset the device, it may be activated later
+ * by your user code. If the update was rejected, just return without doing
+ * anything and we'll wait for another job. If it reported that we should
+ * start test mode, normally we would perform some kind of system checks to
+ * make sure our new firmware does the basic things we think it should do
+ * but we'll just go ahead and set the image as accepted for demo purposes.
+ * The accept function varies depending on your platform. Refer to the OTA
+ * PAL implementation for your platform in aws_ota_pal.c to see what it
+ * does for you.
+ *
+ * @param[in] event Event from OTA lib of type OtaJobEvent_t.
+ * @return None.
+ */
+static void otaAppCallback( OtaJobEvent_t event,
+                            void * pData )
+{
+    OtaErr_t err = OtaErrUninitialized;
+
+    /* OTA job is completed. so delete the MQTT and network connection. */
+    if( event == OtaJobEventActivate )
+    {
+        LogInfo( ( "Received OtaJobEventActivate callback from OTA Agent." ) );
+
+        /* OTA job is completed. so delete the network connection. */
+        MQTT_Disconnect( &mqttContext );
+
+        /* Clear the mqtt session flag. */
+        mqttSessionEstablished = false;
+
+        /* Activate the new firmware image. */
+        OTA_ActivateNewImage();
+
+        /* We should never get here as new image activation must reset the device.*/
+        LogError( ( "New image activation failed." ) );
+
+        for( ; ; )
+        {
+        }
+    }
+    else if( event == OtaJobEventFail )
+    {
+        LogInfo( ( "Received OtaJobEventFail callback from OTA Agent." ) );
+
+        /* Nothing special to do. The OTA agent handles it. */
+    }
+    else if( event == OtaJobEventStartTest )
+    {
+        /* This demo just accepts the image since it was a good OTA update and networking
+         * and services are all working (or we would not have made it this far). If this
+         * were some custom device that wants to test other things before calling it OK,
```

Comment:
Seems like an incomplete phrase in the sentence
"it OK"?

Reply:
"calling it OK" sounds fine to me

Reply:
Ahh, now I see what it means. I think the phrase can be changed to "before determining the new image as validate for use" or something for better clarity.

Reply:
Addressed in https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1418/commits/6fa7ad842095edb46f52d9d38a6e27ffa26a21eb